File: src/1991.find-the-middle-index-in-array.2102/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMiddleIndex = function (nums) {
    const total = nums.reduce((sum, item) => {
        return sum + item;
    }, 0);

    let sum = 0;
    for (let i = 0; i < nums.length; i++) {
        if (sum === total - nums[i] - sum) {
            return i;
        }
        sum += nums[i];
    }

    return -1;
};

File: src/1991.find-the-middle-index-in-array.2102/solution.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int[] rights = new int[nums.length];
        for(int i= nums.length-2;i>-1;i--){
            rights[i] = rights[i+1]+nums[i+1];
        }
        int left = 0;
        for(int i=0;i<nums.length;i++){
            if(left == rights[i]){
                return i;
            }
            left += nums[i];
        }
        
        return -1;
    }
}